src/common/ProfileMenu.tsx

```diff
@@ -0,0 +1,125 @@
+  const { logout } = useAuth()
+
+  let timeout: NodeJS.Timeout
+
+  const copyToClipboard = async () => {
+    try {
+      clearTimeout(timeout)
+      await navigator.clipboard.writeText(`${userDid}`)
+      setIsCopied(true)
+      timeout = setTimeout(() => {
+        setIsCopied(false)
+      }, 3000)
+    } catch (e) {
+      bus.emit(BusEvents['error'], 'Not copied, pleas try again')
+    }
+  }
+
+  const redirectToSupport = () => {
+    window.open(config.SUPPORT_LINK, '_blank', 'noopener noreferrer')
+  }
+
+  const logOut = async () => {
+    await logout()
+    redirect(RoutePaths.SignIn)
+  }
+
+  const userDidLabel = useMemo(() => {
+    if (userDid.length > MAX_LENGTH_DID) {
+      return userDid.substring(0, 8) + '...' + userDid.substring(userDid.length - 4)
+    }
+    return userDid
+  }, [userDid])
+
+  return (
+    <Menu
+      anchorEl={anchorEl}
+      id='account-menu'
+      open={open}
+      onClose={handleClose}
+      anchorOrigin={{ horizontal: 'left', vertical: 'top' }}
+      transformOrigin={{ vertical: 'bottom', horizontal: 'center' }}
+      MenuListProps={{
+        sx: {
+          width: spacing(60.5),
+          border: 1,
+          borderColor: palette.action.hover,
+          borderRadius: spacing(2),
+          boxShadow: shadows[0],
+          pt: 0,
+        },
+      }}
+    >
+      <Stack direction={'column'} p={spacing(4)} alignItems={'center'}>
+        <UserAvatar sx={{ width: spacing(12), height: spacing(12) }} />
+        <Tooltip title={'Copied'} open={isCopied} onClose={handleClose}>
+          <Typography
+            variant={'subtitle4'}
+            overflow={'hidden'}
+            textOverflow={'ellipsis'}
+            maxWidth={spacing(30)}
+            mt={spacing(3)}
+            sx={{ cursor: 'pointer' }}
+            onClick={copyToClipboard}
+          >
+            {userDidLabel}
+          </Typography>
+        </Tooltip>
+      </Stack>
+      <Divider sx={{ py: spacing(2) }} />
+      <MenuItem onClick={handleClose} sx={{ py: spacing(2.5), px: spacing(4) }}>
+        <ListItemIcon>
+          <UiIcon componentName={'qrCode'} size={5} sx={{ color: palette.text.secondary }} />
+        </ListItemIcon>
+        <Typography variant={'caption1'}>QR login</Typography>
+      </MenuItem>
+      <MenuItem onClick={redirectToSupport} sx={{ py: spacing(2.5), px: spacing(4) }}>
+        <ListItemIcon>
+          <UiIcon componentName={'openInNew'} size={5} sx={{ color: palette.text.secondary }} />
+        </ListItemIcon>
+        <Typography variant={'caption1'}>Help Center</Typography>
+      </MenuItem>
+      <MenuItem onClick={logOut} sx={{ py: spacing(2.5), px: spacing(4) }}>
+        <ListItemIcon>
+          <UiIcon componentName={'logOut'} size={5} sx={{ color: palette.error.main }} />
+        </ListItemIcon>
+        <Typography variant={'caption1'} color={palette.error.main}>
+          Disconnect
+        </Typography>
+      </MenuItem>
```

Comment:
1. extract styles to a variable
2. we don't need spacing wrapper for paddings and margins

Reply:
fixed